Modules/ITS/src/ITSDecodingErrorTask.cxx

```diff
@@ -0,0 +1,210 @@
+  getObjectsManager()->startPublishing(mLinkErrorPlots); // mLinkErrorPlots
+  mChipErrorPlots = new TH1D("General/ChipErrorPlots", "Chip Decoding Errors", o2::itsmft::ChipStat::NErrorsDefined, 0.5, o2::itsmft::ChipStat::NErrorsDefined + 0.5);
+  mChipErrorPlots->SetMinimum(0);
+  mChipErrorPlots->SetFillColor(kRed);
+  getObjectsManager()->startPublishing(mChipErrorPlots); // mChipErrorPlots
+}
+
+void ITSDecodingErrorTask::setAxisTitle(TH1* object, const char* xTitle, const char* yTitle)
+{
+  object->GetXaxis()->SetTitle(xTitle);
+  object->GetYaxis()->SetTitle(yTitle);
+}
+
+void ITSDecodingErrorTask::setPlotsFormat()
+{
+  if (mLinkErrorVsFeeid) {
+    setAxisTitle(mLinkErrorVsFeeid, "FeeID", "Error ID");
+  }
+  if (mChipErrorVsFeeid) {
+    setAxisTitle(mChipErrorVsFeeid, "FeeID", "Error ID");
+  }
+  if (mLinkErrorPlots) {
+    setAxisTitle(mLinkErrorPlots, "LinkError ID", "Counts");
+  }
+  if (mChipErrorPlots) {
+    setAxisTitle(mChipErrorPlots, "ChipError ID", "Counts");
+  }
+}
+
+void ITSDecodingErrorTask::startOfActivity(Activity& activity)
+{
+  ILOG(Info, Support) << "startOfActivity : " << activity.mId << ENDM;
+  mRunNumber = activity.mId;
+}
+
+void ITSDecodingErrorTask::startOfCycle() { ILOG(Info, Support) << "startOfCycle" << ENDM; }
+
+void ITSDecodingErrorTask::monitorData(o2::framework::ProcessingContext& ctx)
+{
+  // set timer
+  //
+  std::chrono::time_point<std::chrono::high_resolution_clock> start;
+  std::chrono::time_point<std::chrono::high_resolution_clock> end;
+  // int difference;
+  start = std::chrono::high_resolution_clock::now();
+
+  std::vector<InputSpec> rawDataFilter{ InputSpec{ "", ConcreteDataTypeMatcher{ "DS", "RAWDATA0" }, Lifetime::Timeframe } };
+
+  rawDataFilter.push_back(InputSpec{ "", ConcreteDataTypeMatcher{ "ITS", "RAWDATA" }, Lifetime::Timeframe });
+  DPLRawParser parser(ctx.inputs(), rawDataFilter);
+
+  auto linkErrors = ctx.inputs().get<gsl::span<o2::itsmft::GBTLinkDecodingStat>>("linkerrors");
+  auto decErrors = ctx.inputs().get<gsl::span<o2::itsmft::ChipError>>("decerrors");
+  for (const auto& le : linkErrors) {
+    int istave = (int)(le.feeID & 0x00ff);
+    int ilink = (int)((le.feeID & 0x0f00) >> 8);
+    int ilayer = (int)((le.feeID & 0xf000) >> 12);
+    int ifee = 3 * StaveBoundary[ilayer] - (StaveBoundary[ilayer] - StaveBoundary[NLayerIB]) * (ilayer >= NLayerIB) + istave * (3 - (ilayer >= NLayerIB)) + ilink;
+    for (int ierror = 0; ierror < o2::itsmft::GBTLinkDecodingStat::NErrorsDefined; ierror++) {
+      if (le.errorCounts[ierror] <= 0) {
+        continue;
+      }
+      mLinkErrorCount[ifee][ierror] = le.errorCounts[ierror];
+    }
+  }
+  for (int ifee = 0; ifee < NFees; ifee++) {
+    for (int ierror = 0; ierror < o2::itsmft::GBTLinkDecodingStat::NErrorsDefined; ierror++) {
+      if (mLinkErrorVsFeeid && (mLinkErrorCount[ifee][ierror] != 0)) {
+        mLinkErrorVsFeeid->SetBinContent(ifee + 1, ierror + 1, mLinkErrorCount[ifee][ierror]);
+      }
+    }
+  }
```

Comment:
Just to clarify: are you sure that you can't fill the histogram directly at the 149 line?